src/core/Akka.Streams/Implementation/Fusing/Ops.cs

```diff
@@ -3838,9 +3839,33 @@ public Logic(SourceShape<T> shape, IAsyncEnumerable<T> enumerable) : base(shape)
             public override void PreStart()
             {
                 base.PreStart();
-                _completionCts = new CancellationTokenSource();
                 _enumerator = _enumerable.GetAsyncEnumerator(_completionCts.Token);
             }
+            
+            public override void PostStop()
+            {
+                try
+                {
+                    _completionCts.Cancel();
+                    _completionCts.Dispose();
+                }
+                catch(Exception ex)
+                {
+                    // This should never happen
+                    Log.Error(ex, "AsyncEnumerable threw while cancelling CancellationTokenSource");
+                }
+                try
+                {
+                    var disposed = _enumerator.DisposeAsync().AsTask().Wait(TimeSpan.FromSeconds(10));
```

Comment:
Opted to wait for the dispose task to complete with a generous timeout value (10 seconds).

Reply:
Basically, since the ValueTask was never awaited, we would never catch any exception thrown, since it is ran detached.

Reply:
Reverted back to detached task